src/build_log.cc

```diff
@@ -141,6 +141,28 @@ bool BuildLog::OpenForWrite(const string& path, const BuildLogUser& user,
   return true;
 }
 
+bool BuildLog::RecordLogFile(Edge* edge, const std::string& path, int start_time, int end_time,
+                             TimeStamp mtime) {
+  LogEntry* log_entry = new LogEntry(path);
```

Comment:
Don't new() this, it is never deleted and creates a memory leak. 
I think we can just create a local object. Remember this is C++, not Java ;)